scipy/stats/_fit.py

```diff
@@ -1254,6 +1270,27 @@ def _filliben(dist, data):
 _filliben.alternative = 'less'  # type: ignore[attr-defined]
 
 
+def _order_statistic_evs(dist, n):
+    # Calculate the expected values of the order statistics of a distribution
+    # PDF from https://en.wikipedia.org/wiki/Order_statistic
+    r = np.arange(1, n + 1)
+    constant = gammaln(n + 1) - gammaln(r) - gammaln(n - r + 1)
+    def integrand(x):
+        variable = dist.logpdf(x) + (r-1)*dist.logcdf(x) + (n-r)*dist.logsf(x)
+        return x * np.exp(constant + variable)
+
+    res = integrate.quad_vec(integrand, -np.inf, np.inf)
+    return res[0]
+
+
+def _shapiro_francia(dist, data):
+    n = data.shape[-1]
+    M = _order_statistic_evs(dist, n)
+    X = np.sort(data, axis=-1)
+    return _corr(X, M)**2
```

Comment:
One thing I'm a little unsure about is whether the mean of the sample should be subtracted from the observations in the numerator. This is what happens in `_corr` because it was written for the Filliben test, and the [Filliben statistic is](https://www.jstor.org/stable/1268008):

![image](https://user-images.githubusercontent.com/6570539/224609151-af6097b9-a49e-49e5-81ed-4eada71f5917.png)

But the mean is not subtracted from the $y_i$ in the [original Shapiro-Francia paper](https://www.jstor.org/stable/2284728):

![image](https://user-images.githubusercontent.com/6570539/224609300-7c990d9e-ae18-43f4-b380-a9fbe60c9aa1.png)

[Royston 1983](https://www.jstor.org/stable/2987935) also doesn't subtract the mean:

![image](https://user-images.githubusercontent.com/6570539/224609475-73c9a3d6-d0c8-479e-bc7b-d7f6bb0e0c3a.png)

But in [1992](https://onlinelibrary.wiley.com/doi/abs/10.1002/sim.4780120209?casa_token=NHk9FwYGzhIAAAAA:jJgxkQnqgzIZVV8Vv7FwBWrTTBDzSePzNRs3Zz1vm7m7behHSQTyGVQWfC8mptC_84vXXBI46hP-kQ) he's less explicit:

![image](https://user-images.githubusercontent.com/6570539/224609542-36b86d68-905a-4261-a97b-0b016df9a5af.png)

[The Wikipedia article](https://en.wikipedia.org/wiki/Shapiro%E2%80%93Francia_test) explicitly shows the mean being subtracted:

![image](https://user-images.githubusercontent.com/6570539/224609624-8c45a513-a64b-4099-86c7-51b54c70e03e.png)

But it also shows the Pearson correlation coefficient rather than the square of it.

Based on testing [R's DescTools ShapiroFranciaTest](https://rdrr.io/cran/DescTools/man/ShapiroFranciaTest.html) definitely subtracts the mean. See the `test_against_R_DescTools`; using that code on [rdrr.io](https://rdrr.io/snippets/), I get the same statistic regardless of whether there is a constant offset to the vector or not.

Reply:
Turns out that the statistic is invariant to location and scale regardless of whether the mean is subtracted in the numerator or not.

```python3
import numpy as np
from numpy.testing import assert_allclose
from scipy import stats
rng = np.random.default_rng(1638083107694713882823079058616272161)

loc = 1
scale = 2
x = rng.normal(size=100)

params = {'loc': 0, 'scale': 1}
ref = stats.goodness_of_fit(stats.norm, x*scale + loc, known_params=params,
                            statistic='sf')
res = stats.goodness_of_fit(stats.norm, x, known_params=params,
                            statistic='sf')
print(res.statistic, ref.statistic)  # 0.9843038397823842 0.9843038397823856
np.testing.assert_allclose(res.statistic, ref.statistic)

```